apps/frontend/src/hooks/aave/useAaveReservesHistory.tsx

```diff
@@ -0,0 +1,158 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import axios from 'axios';
+import dayjs from 'dayjs';
+
+import { Decimal } from '@sovryn/utils';
+
+import { AAVE_CONTRACT_ADDRESSES } from '../../constants/aave';
+
+export enum ESupportedTimeRanges {
+  OneMonth = '1m',
+  ThreeMonths = '3m',
+  SixMonths = '6m',
+  OneYear = '1y',
+  TwoYears = '2y',
+  FiveYears = '5y',
+}
+
+export const reserveRateTimeRangeOptions = [
+  ESupportedTimeRanges.OneMonth,
+  ESupportedTimeRanges.SixMonths,
+  ESupportedTimeRanges.OneYear,
+];
+export type ReserveRateTimeRange = typeof reserveRateTimeRangeOptions[number];
+
+type RatesHistoryParams = {
+  from: number;
+  resolutionInHours: number;
+};
+
+type APIResponse = {
+  liquidityRate_avg: number;
+  variableBorrowRate_avg: number;
+  stableBorrowRate_avg: number;
+  utilizationRate_avg: number;
+  x: { year: number; month: number; date: number; hours: number };
+};
+
+const requestCache = new Map<string, Promise<APIResponse[]>>();
+const fetchStats = async (
+  reserveId: string,
+  timeRange: ReserveRateTimeRange,
+  endpointURL: string,
+): Promise<APIResponse[]> => {
+  const { from, resolutionInHours } = resolutionForTimeRange(timeRange);
+  const qs = `reserveId=${reserveId}&from=${from}&resolutionInHours=${resolutionInHours}`;
+  const url = `${endpointURL}?${qs}`;
+
+  if (requestCache.has(url)) {
+    return requestCache.get(url)!;
+  }
+
+  const requestPromise = axios
+    .get<APIResponse[]>(url)
+    .then(response => response.data)
+    .finally(() => {
+      requestCache.delete(url);
+    });
+
+  requestCache.set(url, requestPromise);
+
+  return requestPromise;
+};
+
+const resolutionForTimeRange = (
```

Comment:
it'd better something like this (not tested) : 

``` 
const TIME_RANGES_CONFIG = {
  [ESupportedTimeRanges.OneMonth]: { amount: 30, unit: 'day', resolutionInHours: 6 },
  [ESupportedTimeRanges.SixMonths]: { amount: 6, unit: 'month', resolutionInHours: 24 },
  [ESupportedTimeRanges.OneYear]: { amount: 1, unit: 'year', resolutionInHours: 24 },
};

const calculateUnixTimeFromNow = (amount: number, unit: dayjs.ManipulateType): number => {
  return dayjs().subtract(amount, unit).unix();
};

const resolutionForTimeRange = (
  timeRange: ReserveRateTimeRange,
): RatesHistoryParams => {
  const config = TIME_RANGES_CONFIG[timeRange];

  if (config) {
    return {
      from: calculateUnixTimeFromNow(config.amount, config.unit),
      resolutionInHours: config.resolutionInHours,
    };
  }

  return {
    from: dayjs().unix(), // current time as fallback
    resolutionInHours: 6, // default resolution
  };
};

```

Reply:
@njaume since this piece of code was brought from [Aave](https://github.com/aave/interface/blob/main/src/hooks/useReservesHistory.tsx#L52) 

i prefer to mantain as it is cuz we know its working. do you agree ?